### PR TITLE
fix(api): Fix bug in releases screen where releases don't show up when filtering by environments

### PR DESCRIPTION
**Pull request by wedamija**
This fixes an issue that occurs when a release is associated with an environment via a deploy, but there are no events associated with that release/env combination.

Fixes ISSUE-241

### CODE REVIEW

**Comment by wedamija** [comment]
Not 100% sure this is correct, confirming that the data is as expected in redash

File: src/sentry/api/endpoints/project_releases.py [hunk 1]
```diff
@@ -78,12 +83,10 @@ def get(self, request, project):
...
-                # TODO(LB): May want to change this to ReleaseProjectEnv don't see a
-                # reason to change now.
-                queryset = queryset.filter(id__in=ReleaseEnvironment.objects.filter(
-                    organization_id=project.organization_id,
-                    environment_id=environment.id,
-                ).values_list('release_id', flat=True))
+                queryset = queryset.filter(
+                    releaseprojectenvironment__project=project,
+                    releaseprojectenvironment__environment=environment,
+                )
```
Comment (wedamija) [inline src/sentry/api/endpoints/project_releases.py]: Needs distinct or maybe switch back to using `IN`

Comment (wedamija) [reply src/sentry/api/endpoints/project_releases.py]: Actually it's fine, because we're filtering on all the unique attributes

**Comment by wedamija** [comment]
Spent some time looking at the data. We have some cases where we have a `ReleaseEnvironment` but no corresponding `ReleaseProjectEnvironment`. Not sure how it's happening, because we create them in the same spot in the incoming event ingestion code. Must be related to either deletions or merges, we'll have to investigate further.

File: src/sentry/api/endpoints/project_releases.py (hunk 1 above)
Comment (markstory) [reply src/sentry/api/endpoints/project_releases.py]: :+1: for using a join.

**Review by macqueen — APPROVED** [review]

**Pull request merged**